R/create_target_design.R

```diff
@@ -109,18 +126,18 @@ create_target_design <- function(
     stop("Either targetmin + targetmax or midpoint must be supplied")
   }
   if (!is.null(targetmin) && !is.null(targetmax)) {
-    if (!is_single_valid_number(targetmin) || !is_single_valid_number(targetmax)) {
-      stop("targetmin or targetmax misspecified/not numeric")
+    if (!is.numeric(targetmin) || !is.numeric(targetmax) || length(targetmin) != length(targetmax)) {
+      stop("targetmin or targetmax misspecified/not numeric, or not of same length")
     }
-    midpoint <- mean(c(targetmin, targetmax))
+    midpoint <- apply(cbind(targetmin, targetmax), 1, mean)
```

Comment:
```suggestion
    midpoint <- (targetmin + targetmax) / 2
```
vectorized operation without creation of a new data frame should be more performative, and is also more readable IMO